### DISCUSSION TITLE
Frozen RANS

### DISCUSSION THREAD

**Discussion by JuanSembla** [su2code/SU2#1795] [Q&A]
Hello all! I hope you're doing well. 

I'm working on data-driven turbulence modelling to obtain corrections for the k-omega SST model. I'm using high-fidelity data to feed a machine learning algorithm but I need to compute the specific dissipation rate beforehand. One way of doing so is to use a RANS code to solve the turbulent transport equations with frozen variables and thus get omega. Hence, the values for the velocity field, k and b_ij will remain frozen while the omega and k equations are solved iteratively until we arrive to a converged value for the specific dissipation rate. Evidently, for this to work we need to be able to input the high-fidelity data on SU2 that goes with the mesh and then keep the values frozen while solving the equations. Does anybody know if sth like this could be achieved using SU2?

Thank you in advance. 

Regards.

**Comment by pcarruscag** [comment]
Hi,
In principle yes, you would create a restart file with your values and hardcode relaxation factors of 0 for the flow variables and k.
Whether this gives reasonable values for omega or not I don't know.

**Reply by JuanSembla** [reply]
Hello! Than you so much for your answer, I've been working on it for a few days now. First of all, hardcoding relaxation factors of 0 for the flow variables and k would mean that all the transport equations are computed at each iteration but the values of these variables won't really get updated, right?
I've been trying to find the correct spots to fix the relaxation factors of the flow variables and k to zero. Regarding the flow variables, it would seem that I have to fix the relaxation factor to 0 in the CompleteImplicitIteration_impl function from the CFVMFlowSolverBase.hpp script when adding the solution. For k, I believe it is in the CompleteImplicitIteration function from the CScalarSolver.inl script. Here, I believe the code updates k and omega (which I think are iVar=0 and iVar=1 respectively), so I would need to fix just the one for iVar=0. Is this ok? Are these modifications the correct ones? Thank you in advance.

**Reply by pcarruscag** [reply]
Yes, unless I'm missing some detail that is the way of freezing flow and tke.
However note that as you allow omega to change the residuals of the flow and tke will increase, since the SST blending functions and the turbulent viscosity are functions of omega.

**Reply by JuanSembla** [reply]
Thank you very much! It really worked!
